src/arch/riscv/kernel/vspace.c

```diff
@@ -125,15 +129,51 @@ map_kernel_window(void)
     /* now we should be mapping the kernel base, starting again from PADDR_LOAD */
     assert(pptr == KERNEL_BASE);
     paddr = PADDR_LOAD;
-    /* pptr will overflow at the end of the address range so this loop condition looks odd */
-    while (pptr >= KERNEL_BASE) {
-        assert(IS_ALIGNED(pptr, RISCV_GET_LVL_PGSIZE_BITS(PT_LEVEL_1)));
-        assert(IS_ALIGNED(paddr, RISCV_GET_LVL_PGSIZE_BITS(PT_LEVEL_1)));
 
-        kernel_Level1_pageTable[RISCV_GET_PT_INDEX(pptr, PT_LEVEL_1)] = pte_next(paddr, true);
+#if CONFIG_PT_LEVELS == 3
+    word_t orig_indx_1GiB = RISCV_GET_PT_INDEX(pptr, PT_LEVEL_1);
+#endif
 
-        pptr += RISCV_GET_LVL_PGSIZE(PT_LEVEL_1);
-        paddr += RISCV_GET_LVL_PGSIZE(PT_LEVEL_1);
+    while (pptr < max_address && pptr >= KERNEL_BASE) {
```

Comment:
I believe you can use `UINTPTR_MAX` here instead of max_address.

Reply:
Awesome. I figured there was an easier way. I'll update that and remove the `max_address` calculation